onnxruntime/test/providers/cpu/math/matmul_integer_test.cc

```diff
@@ -312,62 +312,33 @@ void RunMatMulIntegerU8X8Test(const int M, const int N, const int K, bool non_ze
   Eigen::MatrixXi matrix_a = Eigen::MatrixXi::Random(K, M)
                                  .unaryExpr([](int) { return n_unsigned(e); });
   std::vector<uint8_t> matrix_a_data = ToVector<uint8_t>(matrix_a.data(), M * K);
-  uint8_t a_zero_point = non_zero_zp ? GetMiddle(matrix_a_data) : 0;
+  uint8_t a_zero_point =  0;
   Eigen::MatrixXi matrix_a_offset = matrix_a - a_zero_point * Eigen::MatrixXi::Ones(K, M);
 
   Eigen::MatrixXi matrix_b = Eigen::MatrixXi::Random(N, K)
                                  .unaryExpr([](int) { return n_xint8(e); });
   std::vector<WeightType> matrix_b_data = ToVector<WeightType>(matrix_b.data(), N * K);
-  WeightType b_zero_point = non_zero_zp ? GetMiddle(matrix_b_data) : 0;
+  WeightType b_zero_point = 0;
   std::vector<WeightType> b_zp_per_column(N, b_zero_point);
   Eigen::MatrixXi b_zp_matrix = b_zero_point * Eigen::MatrixXi::Ones(N, K);
-  if (non_zero_zp && per_column_zp) {
-    for (int i = 0; i < N; i++) {
-      b_zp_per_column[i] += i % 2 == 0 ? 1 : -1;
-      b_zp_matrix.row(i).setConstant(b_zp_per_column[i]);
-    }
-  }
-
   Eigen::MatrixXi matrix_c = ((matrix_b - b_zp_matrix) * matrix_a_offset).eval();
 
   test.AddInput<uint8_t>("T1", {M, K}, std::move(matrix_a_data));
   test.AddInput<WeightType>("T2", {K, N}, std::move(matrix_b_data), B_is_initializer);
-  if (non_zero_zp) {
-    test.AddInput<uint8_t>("a_zero_point", {}, {a_zero_point});
-    if (per_column_zp) {
-      test.AddInput<WeightType>("b_zero_point", {N}, b_zp_per_column);
-    } else {
-      test.AddInput<WeightType>("b_zero_point", {}, {b_zero_point});
-    }
-  }
 
   test.AddOutput<int32_t>("T3", {M, N}, ToVector<int32_t>(matrix_c.data(), M * N));
-
-  // Nuphar provider does not support non-zero zero point
-  if (non_zero_zp) {
-    test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kNupharExecutionProvider});
-  } else {
-    test.Run();
-  }
+  test.Run();
 }
 
 void RunMatMulIntegerU8X8TestBatch(const int M, const int N, const int K) {
-  RunMatMulIntegerU8X8Test<int8_t>(M, N, K, false /*non_zero_zp*/, false /*B_is_initializer*/, false /*per_column_zp*/);
```

Comment:
why remove those tests?

Reply:
Because when "non_zero_zp", the function will test Nuphar EP exclusively.